code/obj/window.dm

```diff
@@ -176,6 +177,7 @@ ADMIN_INTERACT_PROCS(/obj/window, proc/smash)
 			qdel(src)
 		else if (src.health == 0 && !nosmash)
 			smash()
+		update_icon()
```

Comment:
I wish we had some linter check for this. Basically `update_icon` is the internal proc that you should override and `UpdateIcon` is the external proc that you should call. It does some additional bookkeeping and calls signals that may listen for icon updates.
```suggestion
		UpdateIcon()
```

Reply:
ah yeah I forgot this! thanks!! (for the code review in general 🐝 )